src/averageAges.js

```diff
@@ -1,61 +1,46 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const calculateAverageAge = (persons) => (
+  persons.reduce((sumAges, person) => (
+    sumAges + person.died - person.born
+  ), 0) / persons.length);
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+  const men = people.filter(man => man.sex === 'm');
+  const diedInCentury = person => Math.ceil(person.died / 100) === century;
+
+  return !century
+    ? calculateAverageAge(men)
+    : calculateAverageAge(men.filter(diedInCentury));
```

Comment:
interesting approach, seems you now really understand how callbacks work 😄 

But let's make this return statement work here:

```
return calculateAverageAge(men);
```

A hint: You can select men conditionally inside the filter callback